akka-stream/src/main/scala/akka/stream/BoundedQueueSource.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream
+
+/**
+ * A queue of the given size that gives immediate feedback whether an element could be enqueued or not.
+ */
+trait BoundedQueueSource[T] {
```

Comment:
```
 * Not for user extension
 */
@DoNotInherit
```

Reply:
I agree it doesn't make sense for a user to implement this trait. I followed the convention from the `javadsl|scaladsl.Queue` trait which doesn't have `@DoNotInherit`.

Reply:
I'll add `@DoNotInherit` to `Queue` as well.

Reply:
Looks like all of the mat types don't include this attribute.  I'll just apply it to `BoundedQueueSource`.